hazelcast/src/main/java/com/hazelcast/cache/impl/AbstractCacheService.java

```diff
@@ -164,7 +164,7 @@ private void reset(boolean onShutdown) {
                 if (onShutdown) {
                     partitionSegment.shutdown();
                 } else {
-                    partitionSegment.clear();
+                    partitionSegment.reset();
```

Comment:
I believe [this line](https://github.com/hazelcast/hazelcast/blob/09653c320c4ac4abed2b541391e8faaa5091be92/hazelcast/src/main/java/com/hazelcast/cache/impl/AbstractCacheService.java#L159) in this method can also lead to the journal being destroyed, via this method chain:

`AbstractCacheService#reset(boolean)`
`AbstractCacheService#deleteCache`
`AbstractCacheService#destroySegments`, `AbstractCacheService#closeSegments`
`CachePartitionSegment#deleteRecordStore`
`AbstractCacheRecordStore#close`
`AbstractCacheRecordStore#destroyEventJournal`

Reply:
hmm, right. But currently it's harmless. `AbstractCacheService#reset(boolean)` is called during member shutdown and while resetting services during split-brain merge. In both cases, `RingbufferService` removes all ringbuffers too.

Reply:
I think we eventually need to fix split-brain merge functionality of event-journals. But I haven't touched that part, because I'm not sure but some paths may be used by wan replication too.

Reply:
You're right about split-brain merge for journal. That functionality was postponed until 3.11, along with some other structures:
https://github.com/hazelcast/hazelcast/issues/12244
https://github.com/hazelcast/hazelcast/issues/12245